Curatour/frontend
# Feature/setup pwa

## *Purpose:*
*What does this merge do? Check all that apply.*

## *Description: What does this PR do?*
Added service worker and updated manifest.json to convert application to progressive web application.
Added various links to allow for splash screen to display on iOS devices.
Added function to check online/offline status, and disable certain features while offline.
  - User cannot add event, contact, or agenda item while offline.
Added varying sizes of icons.

## *Instructions for reviewer:*
*Where should the reviewer start?*
`public/index.html`
`public/manifest.json`
`public/sw.js`
`src/components/agenda/AgendaSideBar.js`
`src/components/contacts/Contacts.js`
`src/components/form/Form.js`
`src/index.js`
`src/serviceWorkerRegistration.js`

*How should this be manually tested?*
Open application locally.
Open Chrome devtools and toggle to mobile view.
At the top of the screen where it says "No throttling" click and select "Offline"
Application should still work as normal.
Try to add an event, contact, or agenda item. An error should display.
*Any background context you want to provide?*
![Screen Shot 2021-04-20 at 12 45 20 PM](https://user-images.githubusercontent.com/64617435/115449204-c6c87a80-a1d7-11eb-9fdf-3f2aa8b6f803.png)
![Screen Shot 2021-04-20 at 12 45 02 PM](https://user-images.githubusercontent.com/64617435/115449210-c8923e00-a1d7-11eb-97cd-3c6d6f1a28a0.png)

*What are the relevant tickets?*



## *Issues:*
*Are there any concerns, issues, or bugs in this branch? If so describe them.*
When refreshing the app from the service worker, occasionally an error occurs.
To unregister the service worker and refresh the app, press and hold 'shift' and click the refresh icon on the browser.

## Checklist:
